src/Adapter/Order/CommandHandler/CancelOrderProductHandler.php

```diff
@@ -191,4 +199,24 @@ private function checkOrderState(Order $order)
             );
         }
     }
+
+    /**
+     * @param Order $order
+     */
+    private function cancelOrder(Order $order)
+    {
+        $history = new OrderHistory();
+        $history->id_order = (int) $order->id;
+        $history->changeIdOrderState(Configuration::get('PS_OS_CANCELED'), $order);
+        if (!$history->addWithemail()) {
+            // email failure must not block order update process
+            $this->logger->warning(
+                $this->translator->trans(
+                    'Order history email could not be sent, test your email configuration in the Advanced Parameters > E-mail section of your back office.',
```

Comment:
Hi there, should the 'Advanced Parameters > E-mail' part use placeholders instead? Like we did with issue #18809 with @jolelievre, what do you think?

Reply:
@LouiseBonnard 
Actually I didn't add a new message, I used the same message that was used in the `OrderProductRemover.php` file.

@SD1982  is currently testing this PR so I can't push code right now, but if you want I can change the message in another PR once this one is merged 👍

Reply:
Okay, no worries, it can be done later with, perhaps, other wordings of that kind, it was more a question to Jonathan. But please keep in mind that QA is not supposed to test pull requests when the _waiting for wording_ label is still visible or else it complicates the wording review. :-)

Reply:
yep, I removed the label thinking that since I moving existing code the wording didn't need to be checked again, I'll keep the waiting for wording label to be sure next time 👍